book/routing.rst

```diff
@@ -1550,6 +1550,18 @@ a template helper function:
             Read this blog post.
         </a>
 
+.. tip::
+
+    If you are generating the route inside a script tag, you might need to properly escape it for Javascript
+
+    .. code-block:: javascript
+
+        <script>
+        var route = "{{ path('blog_show', {'slug': 'my-blog-post'})|escape('js') }}";
+        </script>
+
+    For more information, see the Twig documentation.
```

Comment:
I think we should link to the documentation of the `escape` filter here:

```rst
For more information, see the `Twig documentation`_.
```

You then need to add the following to the bottom of the document too:

```rst
.. _`Twig documentation`: http://twig.sensiolabs.org/doc/filters/escape.html
```